skrifa/src/outline/autohint/axis.rs

```diff
@@ -54,6 +59,52 @@ impl Axis {
     }
 }
 
+impl Axis {
+    /// Inserts the given edge into the sorted edge list.
+    ///
+    /// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afhints.c#L197>
+    pub fn insert_edge(&mut self, edge: Edge, top_to_bottom_hinting: bool) {
+        // First push a new default edge
+        self.edges.push(edge);
+        let edges = self.edges.as_mut_slice();
+        // If this is the first edge, we're done.
+        if edges.len() == 1 {
+            return;
+        }
+        // Now move it into place
+        let mut ix = edges.len() - 1;
+        while ix > 0 {
+            let prev_ix = ix - 1;
+            let prev_fpos = edges[prev_ix].fpos;
+            if (top_to_bottom_hinting && prev_fpos > edge.fpos)
+                || (!top_to_bottom_hinting && prev_fpos < edge.fpos)
+            {
+                break;
+            }
+            // Edges with the same position and minor direction should appear
+            // before those with the major direction
+            if prev_fpos == edge.fpos && edge.dir == self.major_dir {
+                break;
+            }
+            let prev_edge = edges[prev_ix];
+            edges[ix] = prev_edge;
+            ix -= 1;
+        }
```

Comment:
It’s just walking downward to find the new location, shifting the elements one at a time until they’re all in place. I think this is basically equivalent to a linear search for insert location followed by a memmove of the elements above? This could probably be a binary search but the predicate is weird and I’m hesitant to deviate too much from FT in code like this.